src/catalog/src/kvbackend/manager.rs

```diff
@@ -339,8 +373,14 @@ impl SystemCatalog {
         }
     }
 
-    fn schema_exists(&self, schema: &str) -> bool {
-        schema == INFORMATION_SCHEMA_NAME || schema == PG_CATALOG_NAME
+    fn schema_exists(&self, schema: &str, catalog_protocol: CatalogProtocol) -> bool {
+        match catalog_protocol {
+            CatalogProtocol::PostgreSQL => schema == PG_CATALOG_NAME,
+            CatalogProtocol::MySQL => schema == INFORMATION_SCHEMA_NAME,
+            CatalogProtocol::Other => {
+                schema == PG_CATALOG_NAME || schema == INFORMATION_SCHEMA_NAME
+            }
+        }
```

Comment:
same as above

Reply:
We can do even harder. Since `pg_catalog` is only visible for pg protocol, we could only build it when we need to start a pg frontend.

Reply:
Sounds reasonable. Can we get postgres options when building system catalog? But since the memory footprint should be small if we don't query it, I think it's acceptable to build it unconditionally.

Reply:
I'will figure it out. If we could, why not?